### PR TITLE
Removed android permissions handling (for android M and above) in favor of PermissionsAndroid library

### PR DESCRIPTION
**Pull request by rajesh1158**

### CODE REVIEW

**Comment by slycoder** [comment]
Hi @rajesh1158 , thanks for putting this together!

In terms of permissions handling, my gut feel is that the application should use the PermissionsAndroid react native library to manage permissions, rather than hard code it into this library.  That way, for example, each application can adjust the message shown to the user to suit their needs and handle rejections differently. 

Just my 2c, feel free to push back.

**Comment by rajesh1158** [comment]
Yeah. I agree. I implemented it this way in my app as a quick solution. We should use PermissionsAndroid library, so this will not be a java change but rather a change in the index.js before calling the connect() function. Correct?

**Comment by bilby91** [comment]
@rajesh1158 Is the work on this PR needed or was done in one of the previous change from @slycoder ?

**Comment by slycoder** [comment]
Hi @rajesh1158, apologies for not responding earlier.  But yes, I imagine that using the PermissionsAndroid library and confining this change to Javascript will be preferable.  It's quite simple to do so so I'm not sure if one even needs us to add any additional code to this library (versus the user of the library adding it to their call site).  But feel free to update this PR and we can see if there are any sticking points with the JS approach that we can smooth out for users of this library.

**Comment by rajesh1158** [comment]
Ok. I will remove the permission related changes from the above code changes.

**Comment by bilby91** [comment]
@rajesh1158 Can you squash your changes please ?

**Comment by rajesh1158** [comment]
Need to read up on 'squashing'. Never done before. Give me some time. Will do it.

**Comment by bilby91** [comment]
@rajesh1158 `git rebase -i abf19c4` and then change the `pick` for `squash` except for the first commit. That will merge all the changes into the one you didn't squashed. 

Hope it helps :)

**Comment by rajesh1158** [comment]
I typed `git rebase -i HEAD~3` and I see below content in nano editor:

pick 783f5d5 added onRequestPermissionsResult override to handle the permissions result
pick 697b40f removed the java changes which were done to support android M permissions
pick af81017 Update README
pick 9e47f63 Update android docs
pick 6959c63 fix(propTypes): add react >16.0.0 support
pick 2df0bec Remove a couple of unused android deps
pick fe59784 Remove unused import
pick e1e2a85 Fix android build for RN 0.47
pick 8f5376c Fix handling of android events
pick a00c8b4 Cleaner syntax
pick 06c0f76 A few switchCamera fixes
pick c7784be Improved camera mirroring behavior on iOS

# Rebase 8db97a4..6bb7640 onto 8db97a4 (12 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out


What do I do?

**Comment by slycoder** [comment]
Here's some more info: 

https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history

**Comment by rajesh1158** [comment]
I think I messed it up. Need some help. What do you guys think about the above rebase / commits I did?

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 1]
```diff
@@ -542,7 +583,7 @@ void pushEvent(View view, String name, WritableMap data) {
         eventEmitter.receiveEvent(view.getId(), name, data);
     }
 
-    public static void registerPrimaryVideoView(VideoView v) {
+	public static void registerPrimaryVideoView(VideoView v) {
```
Comment (slycoder) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java]: This spacing seems funky

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoViewManager.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ protected CustomTwilioVideoView createViewInstance(ThemedReactContext reactConte
     public void receiveCommand(CustomTwilioVideoView view, int commandId, @Nullable ReadableArray args) {
         switch (commandId) {
             case CONNECT_TO_ROOM:
-                String roomName = args.getString(0);
+				String roomName = args.getString(0);
```
Comment (slycoder) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoViewManager.java]: This spacing also seems funky.

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 3]
```diff
@@ -220,23 +254,27 @@ public void onHostDestroy() {
...
          * Create a VideoClient allowing you to connect to a Room
          */
         setAudioFocus(true);
-        ConnectOptions.Builder connectOptionsBuilder = new ConnectOptions.Builder(accessToken);
-		if(roomName != null) {
-			connectOptionsBuilder.roomName(roomName);
-		}
+        ConnectOptions.Builder connectOptionsBuilder = new ConnectOptions.Builder(this.accessToken);
+
+        if(this.roomName != null) {
```
Comment (slycoder) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java]: Could you add a space after the if?

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 4]
```diff
@@ -183,15 +185,47 @@ public void onHostResume() {
...
+                    localVideoTrack.addRenderer(thumbnailVideoView);
+                }
+
+                /*
+                * If connected to a Room then share the local video track.
+                */
+                if (localParticipant != null) {
+                    localParticipant.addVideoTrack(localVideoTrack);
+                }
+                themedReactContext.getCurrentActivity().setVolumeControlStream(AudioManager.STREAM_VOICE_CALL);
```
Comment (slycoder) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java]: You've changed this to only get called when cameraCapturer is not null, but maybe they don't have a camera active but *do* have audio capturing active in which case you'd still want to run this right?

**Comment by slycoder** [comment]
Code looks good.  I think at this point can you go through with the squashing (it looks like you may have squashed the merge in there as well which I think messes things up).

And can you also adjust the title of this PR to reflect the fact that it's no longer about permissions but instead better handling of the app going into the background?

Worst case scenario you can just create a new branch, copy your change, and create a new PR.

**Comment by bilby91** [comment]
@rajesh1158 I agree with @slycoder . Probably the best is to apply the changes from a brand new branch take from current master. I tried to rebase and squash the changes but I'm getting multiple conflicts.

**Comment by rajesh1158** [comment]
Will create a new PR using a new branch from the master to avoid the mess.

**Pull request closed without merging**